ssl/s3_enc.c

```diff
@@ -537,7 +537,7 @@ int ssl3_digest_cached_records(SSL *s, int keep)
         }
 
         /* Loop through bits of algorithm2 field and create MD_CTX-es */
-        for (i = 0; ssl_get_handshake_digest(i, &mask, &md); i++) {
+        for (i = 0; ssl_get_certverify_digest(i, &mask, &md); i++) {
```

Comment:
This is a list of digest to be stored in the tmp.cert_verify_md field. In ssl3_get_cert_verify we select the digest we need according to the pkey we got from the peer.

The ssl_get_handshake_digest can't be used for our purposes because it is used in PRF calculation, and we have only one digest for PRF but a set of them for cert_verify.